src/components/Integrations/BigQuery/AuthorizationPage.js

```diff
@@ -32,13 +37,36 @@ const AuthorizationLayout = ({ dependencies: { bigQueryClient, dopplerUserApiCli
     fetchData();
   }, [bigQueryClient, dopplerUserApiClient]);
 
-  const onSubmit = async (values, emailsToNotify) => {
+  const FieldItemMessage = () => {
+    let message = { delay: successMessageDelay };
+    if (showMessage) {
+      if (errorMessage) {
+        message.text = 'big_query.plus_message_error';
+        message.type = 'cancel';
+      } else {
+        message.text = 'big_query.plus_message_saved';
+        message.type = 'success';
+      }
+    } else {
+      return null;
+    }
+
+    return (
+      <ShowLikeFlash delay={message.delay}>
+        <IconMessage {...message} className="bounceIn" />
+      </ShowLikeFlash>
+    );
+  };
+
+  const onSubmit = async (values) => {
     const emailsData = { emails: [...values.emails] };
     const { success } = await bigQueryClient.saveEmailsData(emailsData);
+    setShowMessage(false);
+    setShowMessage(true);
```

Comment:
I don't understand that lines. 
why does that first change to false and then to true?